contracts/nft-loan/src/execute.rs

```diff
@@ -44,10 +44,17 @@ pub fn deposit_collaterals(
     let borrower = info.sender;
 
     // ensure atleas one asset has been provided
+    // additional check: check if the person is really owner + if approved to be handled by contract
```

Comment:
i agree that adding this check will prevent faulty loans created, how would you recommend to enforce this, as we are not actually expecting any tokens to be transferred until the `accept_loan` and `accept_offer` functions are called, which then do check for correct ownership of the included `AssetInfo's`?

Reply:
Yeah this is how the contract works but I was but proposing to check with `sg721_base::msg::QueryMsg::Approvals` and make sure approval is given to the contract and it expires >= block of the loan so that way when it is accepted then the transfer can actually happens.

But if you make sure that the UI is providing an approval of such, then no need to check that here

Reply:
note: we do not know the actual block of the loan until an offer is accepted or the collateral loan terms is accepted. it will be necessary for  the front end to create the approval msg without an actual set expiration date, and then ensure that the contract will not allow the transfer of the nft if the collateral is withdrawn or the loan is closed

Reply:
```rust
if approval_response.approval.expires > Expiration::AtHeight(env.block.height) {
        return Err(ContractError::Unauthorized {});
    }
```
can be added to enforce that the nft loan is already authorized